homeassistant/components/automation/state.py

```diff
@@ -36,6 +36,16 @@ def state_automation_listener(entity, from_s, to_s):
         @callback
         def call_action():
             """Call action with right context."""
+            if to_s is not None:
```

Comment:
This is not the right solution. Instead, we should not use the state changed helper, but listen to state changed events directly and attach the state_changed event context.